server/src/processing/LinkDetectionModule/LinkDetectionModule.ts

```diff
@@ -29,47 +31,184 @@ import { Module } from '../Module';
 export class LinkDetectionModule extends Module {
   public static moduleName = 'link-detection';
 
-  public main(doc: Document): Document {
-    // actionURI(http://www.axa.com):www.axa.com
-    const actionUriRegex = /(.*)actionURI\((.*)\):(.*)/;
-    // actionGoTo:7,In Real Life
-    const actionGoToRegex = /(.*)actionGoTo:(\d+),(.*)/;
-    // Ref: http://www.cs.cmu.edu/~lemur/doxygen/lemur-3.1/html/Link_8h.html
-    const actionRegex = /(.*)(actionGoToR|actionLaunch|actionNamed|actionMovie|actionUnknown)(.*)/;
+  public async main(doc: Document): Promise<Document> {
+    let mdLinks = await this.extractLinksFromMetadata(doc.inputFile);
+    mdLinks = mdLinks.map((link, id) => ({
+      ...link,
+      id,
+    }));
 
     doc.pages.forEach((page: Page) => {
+      const links = mdLinks.filter(link => parseInt((link as any).page, 10) === page.pageNumber);
+
       page.getElementsOfType<Word>(Word, true).forEach(word => {
-        if (typeof word.content !== 'string') {
-          this.matchLinksInCharacters(word);
-          return;
-        }
-        let match = [];
-        // tslint:disable-next-line:no-conditional-assignment
-        if ((match = word.content.match(actionUriRegex))) {
-          // word.content = `${match[1]}<a href="${match[2]}">${match[3]}</a>`;
-          word.content = match[3];
-          word.properties.link = `${match[1]}<a href="${match[2]}">${match[3]}</a>`;
-          // tslint:disable-next-line:no-conditional-assignment
-        } else if ((match = word.content.match(actionGoToRegex))) {
-          word.content = match[3];
-          word.properties.link = match[1] + match[3];
-          // tslint:disable-next-line:no-conditional-assignment
-        } else if ((match = word.content.match(actionRegex))) {
-          logger.debug('Unknown action: %s', word.content);
+        // for a given word, check if the word matches any not used link position.
+        links.forEach(link => {
+          const l = link as any;
+          const linkBB = new BoundingBox(l.box.l, l.box.t, l.box.w, l.box.h);
+          if (Math.abs(BoundingBox.getPercentageOfInclusion(linkBB, word.box)) > 0.7) {
+            const { link: mdLink, targetURL } = this.buildLinkMD(word, l);
+            word.properties.link = mdLink;
+            word.properties.targetURL = targetURL;
```

Comment:
The value in `targetURL` is understandable, but why does the `mdLink` have to be stored in the Document Representation?
If anything, this value should be generated (using `buildLinkMD()`) inside the markdown exporter.

Reply:
Yes, I started with only the 'link' property and yesterday I added the targetURL to avoid getting it with a RegExp everytime I needed it. 
I will remove 'link' and build it when needed in the exporter